src/functional/ffront/type_info.py

```diff
@@ -89,18 +103,30 @@ def function_signature_incompatibilities_fieldop(
 def function_signature_incompatibilities_scanop(
     scanop_type: ts.ScanOperatorType, args: list[ts.TypeSpec], kwargs: dict[str, ts.TypeSpec]
 ) -> Iterator[str]:
-    if not all(is_field_type_or_tuple_of_field_type(arg) for arg in args):
-        yield "Arguments to scan operator must be fields or tuples thereof."
+    if not all(is_type_or_tuple_of_type(arg, (ts.ScalarType, ts.FieldType)) for arg in args):
+        yield "Arguments to scan operator must be fields, scalars or tuples thereof."
         return
 
-    arg_dims = [extract_dims(el) for arg in args for el in primitive_constituents(arg)]
+    new_args = []
+    new_el: ts.TypeSpec
+    for arg_i in args:
+        if is_type_or_tuple_of_type(arg_i, ts.ScalarType):
+            new_el = apply_to_primitive_constituents(
+                arg_i,
+                lambda primitive_type: ts.FieldType(dims=[], dtype=extract_dtype(primitive_type)),
+            )
+        else:
+            new_el = arg_i
+        new_args.append(new_el)
+
+    arg_dims = [extract_dims(el) for arg in new_args for el in primitive_constituents(arg)]
```

Comment:
```suggestion
```
After your clarification it looks to me that we can remove this and leverage `promote_zero_dims` after we have build the `function_type`. See the suggestion below.

Reply:
Last line is still needed but indeed the rest can be removed. Good call